### ISSUE TITLE
How we can set loss function?

### ISSUE THREAD

**Issue by filipstefanovski** [tensorflow/decision-forests#72]
For example if we need mean absolute error (MAE) , how we can specify as compile parameter doesn't accept loss function?

**Comment by filipstefanovski** [comment]
I found the answer here: https://discuss.tensorflow.org/t/tfdf-custom-loss/2223

**Comment by janpfeifer** [comment]
@filipstefanovski thanks for linking the current best alternative (add the custom loss in the C++ code).

But I'll leave this as an enhancement and leave it open: having a way to wire a TensorFlow defined loss to the training algorithms is one of our outstanding TODOs. It will be slower then defining it in C++, but much more convenient for experimentation and one-of losses.

**Comment by rstz** [comment]
Years later, this can finally be closed :)

- Mean Average Error has been a built-in loss in TF-DF for a while now
- You can train models with custom losses through C++ or with the Python wrapper of YDF. These models can be exported to TF-DF. [Here is the tutorial](https://ydf.readthedocs.io/en/latest/tutorial/custom_loss)
